src/main/java/com/ldtteam/structurize/management/StructureName.java

```diff
@@ -106,13 +108,18 @@ else if (name.startsWith(Structures.SCHEMATICS_CACHE + Structures.SCHEMATICS_SEP
         {
             hut = schematic.split("\\d+")[0];
             section = Structures.SCHEMATICS_PREFIX;
-            if (ForgeRegistries.BLOCKS.getValue(new ResourceLocation(Constants.MINECOLONIES_MOD_ID, "blockhut" + hut)) != Blocks.AIR || HUTS.contains(hut))
+            //This allows mods other than minecolonies to create hut buildings and use the buildtool with them.
+            for (String origin : StructureLoadingUtils.getOriginMods().keySet())
             {
-                section = hut;
+                if (ForgeRegistries.BLOCKS.getValue(new ResourceLocation(origin, "blockhut" + hut)) != Blocks.AIR)
```

Comment:
Again this won't be merged in the current state.
I talked this through with Ray.

Since Structurize 2.0 is a long long long long long way away, we have to work with this.
And since a standing (partially unwritten) rule for this mod is that as long as it is for Minecolonies then everything is allowed. But since PR in particular is targeted at supporting other mods, this won't fly and the exception does not take here.

In other words, create a proper API for this instead of hacking around.

Reply:
I wrote in the other side. Let's remove all the hut stuff from this StructureName class, make in MineColonies an implementation that overrides this Structurename and handle it there.

Reply:
Also possible.